### ISSUE TITLE
Regression: Error when unpacking from io

### ISSUE THREAD

**Issue by erikarvstedt** [msgpack/msgpack-ruby#164]
Reproduce:
```ruby

require 'msgpack'
io = StringIO.new
MessagePack.pack('test', io)
io.rewind
# This throws an error: 'expected Hash but found NilClass' (from unpacker_class.c)
MessagePack.unpack(io)
# This works without an error
MessagePack.unpack(io, {})
```
`MessagePack.unpack(io)` used to work in older versions like 1.0.2. 
To fix this it's probably enough to change the default value of [`param`](https://github.com/msgpack/msgpack-ruby/blob/04012e3e8a8c12f8d4130d212a3d4a90ee78baad/lib/msgpack.rb#L24) to a Hash.

**Comment by tagomoris** [comment]
I confirmed that it's a regression introduced at `v1.2.0`.
I'm preparing a bugfix and release `v1.3.0`.

**Comment by tagomoris** [comment]
I've released `v1.2.8` with fixes for this regression. `v1.2.8` is better than `v1.3.0` for people who specifies `~> 1.2.x`.
